### DISCUSSION TITLE
Proposal for an Improved Design Pattern for Instant Feedback

### DISCUSSION THREAD

**Discussion by andreibarabas** [agrawal-rohit/pearl-ui#213] [Show and tell]
Hi Rohit,

Thanks for putting together this amazing library. As someone that has built design-systems, I was wondering if you encountered the following issue before.

I wrote a proposal for an improved design pattern for handling instant feedback in buttons. I'd love your feedback on it as I'm looking for ways to improve it before posting it later this week. 
https://medium.com/@andreibarabas/rfc-proposal-for-instant-feedback-3dcd4ae4b297

**Comment by agrawal-rohit** [comment]
This is super interesting @andreibarabas! Performance optimization hasn't been my top priority while making Pearl, but this would come in handy when I do start addressing that. Bookmarking this for future reference! 😇

**Reply by andreibarabas** [reply]
Glad you like it. I'm currently exploring the new react native architecture and how it can enable React 18s concurrent features

Here is the first finding, just in case you are curious 

https://medium.com/achieving-100ms/part-1-automatic-batching-w-react-native-fabric-react-18-438ad1282896
